lib/mediator-service/index.js

```diff
@@ -0,0 +1,162 @@
+var CONSTANTS = require('../constants');
+var q = require('q');
+var shortid = require('shortid');
+var MediatorTopicUtility = require('fh-wfm-mediator/lib/topics');
+
+/**
+ * Getting Promises for done and error topics.
+ * This will resolve or reject the returned promise depending on the topic published.
+ *
+ * TODO: This may be of more use in fh-wfm-mediator...
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {Promise}
+ */
+function getTopicPromises(doneTopicPromise, errorTopicPromise) {
+  var deferred = q.defer();
+
+  doneTopicPromise.then(function(response) {
+    deferred.resolve(response);
+  });
+
+  errorTopicPromise.then(function(error) {
+    deferred.reject(error);
+  });
+  return deferred.promise;
+}
+
+/**
+ * A mediator service that will publish and subscribe to topics to be able to render file data.
+ *
+ * @param {Mediator} mediator
+ * @param {object}   config
+ * @constructor
+ */
+function FileMediatorService(mediator, config) {
+  this.mediator = mediator;
+  this.config = config || {};
+  
+  this.usersTopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.USERS_ENTITY_NAME);
+  this.filesTopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.FILES_ENTITY_NAME);
+  this.fileUITopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.FILE);
```

Comment:
@wtrocki should this not be using the `FILE_TOPIC_UI_PREFIX`? https://github.com/wtrocki/raincatcher-file-angular/blob/68ec0c752cd0edcdacdaeee97c853c5b8d1da4c2/lib/constants.js#L7